src/graphql/node/control.rs

```diff
@@ -39,188 +38,249 @@ impl NodeControlMutation {
+                node.profile,
+                node.profile_draft,
+            );
+
+            broadcast_customer_change_if_needed(ctx, &node).await;
+        }
 
-            if let Some(customer_id) = should_broadcast_customer_change(&node) {
-                broadcast_customer_change(customer_id, ctx).await?;
+        if let Some(ref target_agents) = apply_scope.agents {
+            let agent_manager = ctx.data::<BoxedAgentManager>()?;
+            if let Err(e) = notify_agents(
+                agent_manager,
+                &target_agents.updates,
+                &target_agents.disables,
+            )
+            .await
+            {
+                error!("Failed to notify agents for node {i} to be updated. However, the failure does not affect the node application operation.\nDetails:\n{e:?}",);
             }
-            config_setted_modules
-        } else {
-            return Err("Failed to apply node profile".into());
-        };
 
-        Ok(ApplyResult {
-            id,
-            success_modules,
-        })
+            info!(
+                "[{}] Node ID {i} - Node's agents are notified to be updated.\n{:?}",
+                chrono::Utc::now(),
+                node.agents.iter().filter_map(|agent| {
+                    if target_agents.updates.contains(&agent.key.as_str()) {
+                        Some(format!(
+                            "\nAgent key: {}, Config: {:?}, Draft: {:?}",
+                            agent.key, agent.config, agent.draft
+                        ))
+                    } else {
+                        None
+                    }
+                })
+            );
+        }
+
+        Ok(ApplyNodeResponse { id, giganto_draft })
     }
 }
 
 #[derive(SimpleObject, Clone)]
-pub struct ApplyResult {
+pub struct ApplyNodeResponse {
+    /// The ID of the node to which the draft was applied.
     pub id: ID,
-    pub success_modules: Vec<ModuleName>,
+
+    /// The draft of the Giganto module, associated with the node.
+    ///
+    /// If `None`, it means either the node does not have the Giganto module or the draft for the
+    /// Giganto module is not available. In the latter case, this indicates that the Giganto should
+    /// be disabled.
+    pub giganto_draft: Option<String>,
 }
 
-async fn send_set_config_requests(
-    agents: &BoxedAgentManager,
-    node: &Node,
-) -> anyhow::Result<Vec<ModuleName>> {
-    let profile_draft = node
-        .profile_draft
-        .as_ref()
-        .ok_or_else(|| anyhow::anyhow!("There is nothing to be applied."))?;
-
-    let mut result_combined: Vec<ModuleName> = vec![];
-
-    for (module_name, config) in target_app_configs(profile_draft) {
-        if send_set_config_request(
-            agents,
-            &profile_draft.hostname,
-            module_name.as_ref(),
-            &config,
-        )
-        .await?
-        {
-            result_combined.push(module_name);
-        }
-    }
+async fn notify_agents(
```

Comment:
I don't think the name `notify_agents` is quite fitting because this function not only notifies the agents but also sends the new configuration and updates them.

Reply:
Sorry for the confusion. I should've posted this in the `applyNode` PR.